test/robot_test.coffee

```diff
@@ -35,6 +35,7 @@ describe 'Robot', ->
 
     # Re-throw AssertionErrors for clearer test failures
     @robot.on 'error', (name, err, response) ->
+      return unless err?
```

Comment:
I guess I ran into this before, but solved [a slightly different way](https://github.com/github/hubot/blob/receive-middleware/test/middleware_test.coffee#L291). I think it will handle an additional edge case, but I don't know how much it matters. I have absolutely no memory of that though, so who knows.